### DISCUSSION TITLE
In Prisma 4.6.0, changing `onDelete: SetNull` to `Cascade` results in recreating foreign key

### DISCUSSION THREAD

**Discussion by jameyhart** [prisma/prisma#16223] [Q&A] [answered]
@janpio After upgrading to 4.6.0, we started reciving the following error when migrating our database:
```sh
Error: Schema validation error - Error (query-engine-node-api library)
Error code: P1012
error: Error parsing attribute "@relation": The `onDelete` referential action of a relation must not be set to `SetNull` when a referenced field is required.
Either choose another referential action, or make the referenced fields optional.

  -->  schema.prisma:126
   |
125 |     
126 |     survey    Survey   @relation(fields: [surveyId], references: [id], onDelete: SetNull)
127 |
   |
```

To rectify this, we changed the `onDelete` from `SetNull` to `Cascade` but it then cerates a new migration that drops the foreign key and recreates it.
Is this the expected behaviour?

Our current database model:
```prisma
model Survey {
    id                      String                    @id @default(cuid())
    name                    String
    survey                  Json
    reportSurvey            SurveyReport[]
    ...
}

model SurveyReport {
    id                String       @id @default(cuid())
    surveyId          String
    survey            Survey       @relation(fields: [surveyId], references: [id], onDelete: SetNull)
    reportType        SurveyReportType
    ...
}
```

The output from the new migration:
```sql
-- DropForeignKey
ALTER TABLE "SurveyReport" DROP CONSTRAINT "SurveyReport_surveyId_fkey";

-- AddForeignKey
ALTER TABLE "SurveyReport" ADD CONSTRAINT "SurveyReport_surveyId_fkey" FOREIGN KEY ("surveyId") REFERENCES "Survey"("id") ON DELETE CASCADE ON UPDATE CASCADE;
```

**Comment by janpio** [comment, marked as answer]
Continued here: https://github.com/prisma/prisma/issues/16228

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
